ddtrace/internal/writer.py

```diff
@@ -116,14 +121,55 @@ def __repr__(self):
         )
 
 
-class LogWriter:
+class TraceWriter(six.with_metaclass(abc.ABCMeta)):
```

Comment:
This ABC seems to implement a good chunk of the `Thread` API, which seems inspired by the concrete case of the `AgentWriter`. Would it make sense to stipulate that a `TraceWriter` is effectively a thread, given that any potential concretions will perform some form of I/O? In this regard, wouldn't it be best for the `LogWriter` to do I/O in a separate thread as well?

Reply:
I actually don't think it needs to be a thread. It *can* be a thread, but you could have a totally synchronous writer AFAICT.

So the API for a writer should be just `write` and `recreate`.

Reply:
Yeah I tried to outline what the current API is (what is used/assumed in `Tracer`). I definitely don't think that this is what the ideal API should look like (eg. having all the thread methods).

Reply:
Fair enough, but is there a good reason to not set a correct API right now? 😄

Reply:
```python

class TraceWriter:
    pass

class ThreadWriter(TraceWriter):
    pass

class LogWriter(ThreadWriter):
    pass
```

?

Reply:
Sure, we can make the implementation changes before introducing this base class